src/main.rs

```diff
@@ -208,15 +210,21 @@ impl DWebBackend {
         let kind = Some(CRYPTO_KIND_VLD0);
 
         let dht_record = routing_context.create_dht_record(schema, kind).await?;
+        let keypair = vld0_generate_keypair();
         let encryption_key = CryptoTyped::new(CRYPTO_KIND_VLD0, CryptoKey::new([0; 32]));
 
         let group = Group {
-            id: encryption_key.value,
+            id: keypair.key.clone(),
             dht_record,
             encryption_key,
+            secret_key: CryptoTyped::new(CRYPTO_KIND_VLD0, keypair.secret),
```

Comment:
Lets make the group secret optional as well  so that we can have peers that can read from a group but can't add themselves to it.